tracer/src/Datadog.Trace.Tools.Runner/Checks/ProcessBasicCheck.cs

```diff
@@ -81,9 +84,52 @@ public static bool Run(ProcessInfo process)
                 foundIssue = true;
             }
 
+            if (!CheckRegistry())
+            {
+                foundIssue = true;
+            }
+
             return !foundIssue;
         }
 
+        internal static bool CheckRegistry(IRegistryService? registry = null)
+        {
+            registry ??= new Windows.RegistryService();
+
+            try
+            {
+                var suspiciousNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+                {
+                    "COR_ENABLE_PROFILING",
+                    "CORECLR_ENABLE_PROFILING",
+                    "COR_PROFILER",
+                    "CORECLR_PROFILER",
+                    "COR_PROFILER_PATH",
+                    "CORECLR_PROFILER_PATH",
+                    "COR_PROFILER_PATH_64",
+                    "CORECLR_PROFILER_PATH_64"
+                };
+
+                bool foundKey = false;
+
+                foreach (var name in registry.GetLocalMachineValueNames(@"SOFTWARE\Microsoft\.NETFramework"))
+                {
+                    if (suspiciousNames.Contains(name))
+                    {
+                        Utils.WriteWarning(SuspiciousRegistryKey(name));
```

Comment:
Just because _we_ don't set them, doesn't mean it's not valid for the customer to set them? I think we should probably validate them as best we can (e.g. COR_PROFILER, COR_ENABLE_PROFILING etc) and just flag the ones we can't be sure are right (path based)?

Reply:
Honestly I don't think it's worth the extra complexity. I find it extremely unlikely that a customer would declare those variables in a way that is not described by our documentation.